PTGLgraphComputation/src/proteingraphs/ComplexGraph.java

```diff
@@ -187,31 +231,43 @@ public ComplexGraph(String pdbid, List<Chain> chains, List<MolContactInfo> resCo
     
     /**
      * Creates the vertices, fills corresponding maps and fills amino acid sequence.
-     * @param chains All chains of this complex graph
+     * @param chains All chains of this complex graph. Hint: these chains are not the same objects as the ones that are parsed
      */
     private void createVertices(List<Chain> chains) {
+        ArrayList<Chain> allChains = FileParser.getChains();    // since the input chains are not the same objects as the parsed chains and therefore do not have all their attributes,
+        Vertex v;                                               // we need to compare the input chains to the parsed chains to find out if they are RNA
         for(Integer i = 0; i < chains.size(); i++) {
             Chain tmpChain = chains.get(i);
-            Vertex v = createVertex();
+            for (Integer j = 0; j < allChains.size(); j++) {    // loop through chains to find matches
+                if (tmpChain.getPdbChainID() == allChains.get(j).getPdbChainID()) {     // if the chains match, check their molecule type
+                    if (allChains.get(j).getMoleculeType().contains("polyribonucleotide") && ! includeRna) {  // hint: if including of RNA is turned off, RNA/DNA hybrids are not included either
+                        System.out.println("continue");
+                        continue;
+                    }
+                    else {
+                        v = createVertex();
```

Comment:
This looks extremely ugly. First, ComplexGraph should not get the chains from the file parser but receive them via the constructor from main instead, for example. Second, looping through both chain lists to find the matches ideally should not be necessary. Also runtime-wise this is bad. 
But I guess if it works, you should not be the one to tidy up the strange architecture. I still wanted to mention it.

Reply:
Yes I thought so too. But at least runtime-wise it is not going to matter that much in practice, since most molecules do not have that many chains. But still.
At least I fixed the FileParser thing.